qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -61,4 +90,220 @@ public void test10Install() {
+        assumeThat(installation, is(notNullValue()));
+
+        final Path tempConf = getTempDir().resolve("esconf-alternate");
+
+        try {
+            mkdir(tempConf);
+            cp(installation.config("elasticsearch.yml"), tempConf.resolve("elasticsearch.yml"));
+            cp(installation.config("log4j2.properties"), tempConf.resolve("log4j2.properties"));
+
+            // we have to disable Log4j from using JMX lest it will hit a security
+            // manager exception before we have configured logging; this will fail
+            // startup since we detect usages of logging before it is configured
+            final String jvmOptions =
+                "-Xms512m\n" +
+                "-Xmx512m\n" +
+                "-Dlog4j2.disable.jmx=true\n";
+            append(tempConf.resolve("jvm.options"), jvmOptions);
+
+            final Shell sh = new Shell();
+            if (Platforms.WINDOWS) {
+                sh.powershell(
+                    "$account = New-Object System.Security.Principal.NTAccount 'vagrant'; " +
+                    "$tempConf = Get-ChildItem '" + tempConf + "' -Recurse; " +
+                    "$tempConf += Get-Item '" + tempConf + "'; " +
+                    "$tempConf | ForEach-Object { " +
+                        "$acl = Get-Acl $_.FullName; " +
+                        "$acl.SetOwner($account); " +
+                        "Set-Acl $_.FullName $acl " +
+                    "}"
+                );
+            } else {
+                sh.bash("chown -R elasticsearch:elasticsearch " + tempConf);
+            }
+
+            final Shell serverShell = new Shell();
+            serverShell.getEnv().put("ES_PATH_CONF", tempConf.toString());
+            serverShell.getEnv().put("ES_JAVA_OPTS", "-XX:-UseCompressedOops");
+
+            Archives.runElasticsearch(installation, serverShell);
+
+            final String nodesResponse = makeRequest(Request.Get("http://localhost:9200/_nodes"));
+            assertThat(nodesResponse, containsString("\"heap_init_in_bytes\":536870912"));
+            assertThat(nodesResponse, containsString("\"using_compressed_ordinary_object_pointers\":\"false\""));
+
+            Archives.stopElasticsearch(installation);
+
+        } finally {
+            rm(tempConf);
+        }
+    }
+
+    @Test
+    public void test80RelativePathConf() {
+        assumeThat(installation, is(notNullValue()));
+
+        final Path temp = getTempDir().resolve("esconf-alternate");
+        final Path tempConf = temp.resolve("config");
+
+        try {
+            mkdir(tempConf);
+            Stream.of(
+                "elasticsearch.yml",
+                "log4j2.properties",
+                "jvm.options"
+            ).forEach(file -> cp(installation.config(file), tempConf.resolve(file)));
+
+            append(tempConf.resolve("elasticsearch.yml"), "node.name: relative");
+
+            final Shell sh = new Shell();
+            if (Platforms.WINDOWS) {
```

Comment:
This conditional is repeated a few times and not very readable, I might consider extracting it to a utility method. 
Would also consider adding `Platform.onWindowsOnly(PlatformSpecificAction do, String description)` where `PlatformSpecificAction` is just a functional interface,  I would find that more readable, and besides the `else` condition is not technically correct java  as not windows doesn't imply posix ( in theory at least ). It would also give us the opportunity  to  log that `desciption` was carried out, or it wasn't on thi platform.

Reply:
I've mostly tried to keep the control flow as simple as possible in cases like this where we're very unlikely to need to extend it any further - that is, even though the `if (windows) {} else {}` is gross, the effect is really clear and it's unlikely to ever need a third code path.

I like what you've suggested here though and I'll try it out.

>the else condition is not technically correct java as not windows doesn't imply posix ( in theory at least )

Agree, this has consistently bothered me. Similarly though, making more of a distinction than that has seemed like overkill

Reply:
Yes, it would have been an overkill and useless distraction in a conditional, that's also why I suggested abstracting the platform specifics a bit more. Assuming that the platform abstractions are correct, most of the test code doesn't care about the platform, having these conditionals only when there is an actual difference in behavior between these platforms that we need to test for would make it a much easier read and imho help allot in maintaining these tests.